### PR TITLE
GLK: ZCODE: detection_tables.h Mass Update

### PR DESCRIPTION
**Pull request by MarcoBorrini99**
This PR is an attempt to document a large amount of missing zcode title with info taken from different sources scattered all around the Internet.

It focuses on two different areas:

1. Updates entries from different sources:

- https://ifdb.org/search?searchfor=system%3AInform+5
- https://ifdb.org/search?searchfor=system%3AInform+6
- https://ifdb.org/search?searchfor=system%3AInform+7
- http://www.ifarchive.org/if-archive/
- https://eblong.com/infocom/
- http://ifiction.free.fr/
- http://www.fiction-interactive.fr/category/fictions-interactives/
- http://textadventures.co.uk/
- https://playfic.com/explore/name (usually medium/low level quality titles)

2. Updates entries for all possible competitions

### CODE REVIEW

**Comment by MarcoBorrini99** [comment]
Suggestions / Improvements?

File: engines/glk/zcode/detection_tables.h [hunk 1]
```diff
@@ -93,1108 +104,2715 @@ const PlainGameDescriptor ZCODE_GAME_LIST[] = {
 	{ "zcode", "Unknown Z-code game" },
 
 	// English games
-	{ "404life",           "404 - Life not found" },
```
Comment (sev-) [inline engines/glk/zcode/detection_tables.h:96]: This must be reverted.

Comment (MarcoBorrini99) [reply engines/glk/zcode/detection_tables.h:96]: Got it.
Must this identation be kept only for **// English games**, **// Painfull Little Stupid Games** and **// Converted Scott Adams** games sections?

By the way, I've also sorted some sections (**// French games** for example) while adding new titles. Do I also need to restore the original scheme?

Comment (sev-) [reply engines/glk/zcode/detection_tables.h:96]: If by sorting you mean alphabetical sorting, yes, that is a good thing. However, please just do not alter the whitespaces.

Comment (MarcoBorrini99) [reply engines/glk/zcode/detection_tables.h:96]: Thanks @sev- 

Just added a commit to revert old identation.

BTW I'd like to explain what I have in my mind. 
For my little PI400 I'm building a scummvm libretro installation focused on old adventures (mainly Inform ones).

I'd like to apply zcode documentation updates in detection_tables.h to "renew" other GLK subengines with the same style.

So I kindly ask some of your time for these questions:

- No tab character, only whitespaces. Right?
- Why some sections have different identations? For consistency, I think that the same one should always be used. Do you agree to apply the same used for "//English games"?
- I have documented a lot of competitions entries but each competition usually covers a lot of different GLK subsystem so I've choosen to prepended each entry with a prefix specific to that competition. What do you think?

Example:
ZCODE "Metamorphoses" and TADS "Jarod's Journey" in "The 2000 Annual Interactive Fiction Competition" are documented as

engines/glk/zcode/detection_tables.h
```
// The 2000 Annual Interactive Fiction Competition
{ "if00_metamorphoses", "Metamorphoses" },
```
[...]

engines/glk/tads/detection_tables.h
```
// The 2000 Annual Interactive Fiction Competition
{ "if00_jarodsjourney", "Jarod's Journey" },
```
[...]

Comment (sev-) [reply engines/glk/zcode/detection_tables.h:96]: > * No tab character, only whitespaces. Right?

If you're talking about the alignment space for vertical columns, then both tabs are spaces are allowed. Tabs are 4 spaces wide. If you're talking about the indentation, then it is tabs-only.

> * Why some sections have different identations? For consistency, I think that the same one should always be used. Do you agree to apply the same used for "//English games"?

Historical reasons and to make the total width minimal


> * I have documented a lot of competitions entries but each competition usually covers a lot of different GLK subsystem so I've choosen to prepended each entry with a prefix specific to that competition. What do you think?

I like it!

I see no further issues with your changes, merging.

**Comment by sev-** [comment]
I see that there are non-compilable commits. Could you please make sure that every commit is compilable?

**Comment by MarcoBorrini99** [comment]
I decided to close this PR after I seriuosly broke my local git copy and I don't want to pollute the correct git commit messages flow...

**Pull request closed without merging**